sdk/src/main/java/com/mongodb/stitch/android/auth/DecodedJWT.java

```diff
@@ -0,0 +1,59 @@
+package com.mongodb.stitch.android.auth;
+
+import android.util.Base64;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mongodb.stitch.android.StitchException;
+
+import java.io.IOException;
+import java.util.Map;
+
+@SuppressWarnings("WeakerAccess")
+public class DecodedJWT {
+
+    private final String[] parts;
+    private final Map<String, Object> payload;
+
+    public DecodedJWT(final String jwt) {
+        parts = splitToken(jwt);
```

Comment:
[rec] Seems like once we've mapped the payload to our field, we no longer need the `parts` field to be stored. It would be more useful for a user to have the raw jwt saved as a field as such:

```java
public final String raw;
...
public DecodedJWT(final String jwt) {
    this.raw = jwt;
    ...
}
```